devito/types/array.py

```diff
@@ -441,12 +443,13 @@ def initvalue(self):
 
     # Overrides defaulting to self.c0's behaviour
 
-    for i in ['_mem_internal_eager', '_mem_internal_lazy', '_mem_local',
+    for i in ('_mem_internal_eager', '_mem_internal_lazy', '_mem_local',
               '_mem_mapped', '_mem_host', '_mem_stack', '_mem_constant',
               '_mem_shared', '__padding_dtype__', '_size_domain', '_size_halo',
               '_size_owned', '_size_padding', '_size_nopad', '_size_nodomain',
               '_offset_domain', '_offset_halo', '_offset_owned', '_dist_dimensions',
-              '_C_get_field', 'grid', 'symbolic_shape']:
+              '_C_get_field', 'grid', 'symbolic_shape',
+              *AbstractFunction.__properties__):
```

Comment:
Nitpick: `('_mem_internal_eager', '_mem_internal_lazy', ...) + AbstractFunction.__properties__` might be tidier imo

Reply:
`a + b` creates two tuples a, b then a new third one to merge them while current one only creates one directly.